src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,70 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+    private static final int THRESHOLD = 500;
+
+    private final int[] array;
+    private final int start;
+    private final int end;
 
     public MergeSortAction(int[] array) {
+        this(array, 0, array.length);
+    }
+
+    public MergeSortAction(int[] array, int start, int end) {
+        this.array = array;
+        this.start = start;
+        this.end = end;
+    }
+
+    @Override
+    protected void compute() {
+        if (end - start <= THRESHOLD) {
+            sequentialSort(array, start, end);
+        } else {
+            int mid = (start + end) / 2;
```

Comment:
Calculating the mid-point using (start + end) / 2 could potentially cause an integer overflow if start and end are large enough. To prevent this, use int mid = start + (end - start) / 2 instead.